lib/fact_changes.rb

```diff
@@ -1,32 +1,28 @@
 require 'token_util'
 require 'changes_support/disjoint_list'
 require 'changes_support/transaction_scope'
-require 'google_hash'
 
 class FactChanges
+  ACTIONS = [
+    'set_errors', 'create_assets', 'create_asset_groups', 'delete_asset_groups',
+    'remove_facts', 'add_facts', 'delete_assets', 'add_assets', 'remove_assets'
+  ].freeze
+
   include ChangesSupport::TransactionScope
 
   attr_accessor :facts_to_destroy, :facts_to_add, :assets_to_create, :assets_to_destroy,
-    :assets_to_add, :assets_to_remove, :wildcards, :instances_from_uuid,
-    :asset_groups_to_create, :asset_groups_to_destroy, :errors_added,
-    :already_added_to_list, :instances_by_unique_id,
-    :facts_to_set_to_remote
-
-  attr_accessor :operations
-
+                :assets_to_add, :assets_to_remove, :wildcards, :instances_from_uuid,
+                :asset_groups_to_create, :asset_groups_to_destroy, :errors_added,
+                :already_added_to_list, :instances_by_unique_id,
+                :facts_to_set_to_remote, :operations
 
   def initialize(json = nil)
-    @assets_updated=[]
+    @assets_updated = []
     reset
-    parse_json(json) if json
-  end
-
-  def parsing_valid?
```

Comment:
Removing this as its not used, and in-practice we'll almost always be throwing an exception with invalid json.